src/AzureExtension/AzureExtension.cs

```diff
@@ -30,6 +34,8 @@ public AzureExtension(ManualResetEvent extensionDisposedEvent)
                 return new RepositoryProvider();
             case ProviderType.FeaturedApplications:
                 return new object();
+            case ProviderType.ComputeSystem:

```

Comment:
I'm interested in how this'll work.  Returning a new object with another object inside of it, across a DLL boundary.

Reply:
This PR is actually the full PR stripped of the implementation so that it can be reviewed in parts. It currently works end to end.

Reply:
Is there a better way to do this?

Reply:
I believe what Darren is saying here is that we're doing `new Object1( Another object that Object1 does not control) ` So it looks like there is no way for Object1 (who we pass to Dev Home) to know the lifetime of the other object. However, the _host objects lifetime will live for as long as the extension is running. The other question he is asking here is what is the lifetime of Object1? we just create an instance and send it off to dev home. While I'm not 100% sure who would own the lifetime of this new object since we create the instance and send it off to Dev Home (cross process). I do believe its probably Dev Home at this point and Object1 should be a singleton especially if we expect Object1 to contain instance specific data that should be the same after every call here. 

so, the DevBoxProvider should be a singleton in this case. Similar to what DeveloperId is doing, but since were using a Host container, we could use the container to get the singleton for DevBoxProvider.

Reply:
I agree, it should be a singleton. Technically we could have multiple instances, but they would be doing the same work, potentially introducing unnecessary concurrency.
The lifetime should be owned by the client (DevHome). Until all references released the object should not be destructed.